src/components/timeline/TimelineViewControls.svelte

```diff
@@ -0,0 +1,143 @@
+<script lang="ts">
+  import ArrowLeftIcon from '@nasa-jpl/stellar/icons/arrow_left.svg?component';
+  import ArrowRightIcon from '@nasa-jpl/stellar/icons/arrow_right.svg?component';
+  import MinusIcon from '@nasa-jpl/stellar/icons/minus.svg?component';
+  import PlusIcon from '@nasa-jpl/stellar/icons/plus.svg?component';
+  import RotateCounterClockwiseIcon from '@nasa-jpl/stellar/icons/rotate_counter_clockwise.svg?component';
+  import { onMount } from 'svelte';
+  import { maxTimeRange, viewTimeRange } from '../../stores/plan';
+  import { tooltip } from '../../utilities/tooltip';
+
+  export let nudgePercent = 0.05;
+
+  $: maxDuration = $maxTimeRange.end - $maxTimeRange.start;
+  $: viewDuration = $viewTimeRange.end - $viewTimeRange.start;
+
+  onMount(() => {
+    document.addEventListener('keydown', onKeydown);
+
+    return () => {
+      document.removeEventListener('keydown', onKeydown);
+    };
+  });
+
+  function onKeydown(e: KeyboardEvent & { target: HTMLInputElement }) {
+    // If user holds shift while not focused on an input then activate the temporary unlock.
+    // If an input is focused, we assume they're holding shift to capitalize instead.
+    if (e.target.tagName !== 'INPUT') {
+      if (e.key === '=') {
+        onZoomIn();
+      } else if (e.key === '-') {
+        onZoomOut();
+      } else if (e.key === '[') {
+        onNudgeLeft();
+      } else if (e.key === ']') {
+        onNudgeRight();
+      } else if (e.key === '0') {
+        onResetViewTimeRange();
+      }
+    }
+  }
+
+  function getViewTimeRangePercentZoom() {
+    return ($viewTimeRange.end - $viewTimeRange.start) / ($maxTimeRange.end - $maxTimeRange.start);
+  }
+
+  function onZoomIn() {
+    // Compute current zoom percentage
+    const percentZoom = getViewTimeRangePercentZoom();
+    let newDuration = Math.max((percentZoom - 0.05) * maxDuration, 60000); // Min zoom of one minute
+
+    const pivotTime = $viewTimeRange.start + viewDuration / 2;
+    const newStart = pivotTime - newDuration / 2;
+    const newEnd = pivotTime + newDuration / 2;
+    $viewTimeRange = { end: newEnd, start: newStart };
+  }
+
+  function onZoomOut() {
+    // Compute current zoom percentage
+    const percentZoom = getViewTimeRangePercentZoom();
+    let newDuration = (percentZoom + 0.05) * maxDuration;
+
+    // Clamp zoom
+    if (viewDuration >= maxDuration) {
+      newDuration = maxDuration;
+      $viewTimeRange = $maxTimeRange;
+      return;
+    }
+
+    const pivotTime = $viewTimeRange.start + viewDuration / 2;
+    const newStart = Math.max(pivotTime - newDuration / 2, $maxTimeRange.start);
+    const newEnd = Math.min(pivotTime + newDuration / 2, $maxTimeRange.end);
+    $viewTimeRange = { end: newEnd, start: newStart };
+  }
+
+  // function onShiftLeft() {
```

Comment:
Remove commented out functions

Reply:
Will do, waiting to chat with Lyle to see if this behavior might be desired